homework_4/Barbhaiwala/solution.js

```diff
@@ -0,0 +1,73 @@
+var colors = require("colors");
+
+function HotelParser(jsonString) {
+    
+    this.hotelInfoJson = jsonString; 
+    
+    this.parse = function () { 
+        var parsedHotelInfo = parseHotelInfoFromJson();
+        
+        var hotelList = verify(parsedHotelInfo.hotelList);
+        return hotelList;
+    }
+    
+    function verify(hotelList) {
+        var hotelInformation = [];
+        
+        if (!hotelList) {
+            return hotelInformation;
+        }
+        //for (var hotel in hotelList) {
+        //hotelList.foreach(function (hotel) {
+        for(var idx = 0, hotel; idx < hotelList.length; ++idx) {
```

Comment:
hotel won't be scoped to the for, but to the `verify()` function, so I would suggest declaring it above. Putting it here might make some people assume that it isn't accessible outside the `for`.